### ISSUE TITLE
E666: compiler not supported: asciidoctor-pdf

### ISSUE THREAD

**Issue by KingJulianS** [habamax/vim-asciidoctor#108]
Hi again, apologies for these newb issues. 

I am getting the above error msg when sourcing my vimrc.  
Any ideas ?
   
```
    Plug 'habamax/vim-asciidoctor'
     
    let g:asciidoctor_executable = 'asciidoctor'
    "let g:asciidoctor_extensions = ['asciidoctor-diagram', 'asciidoctor-rouge']
    let g:asciidoctor_pdf_executable = 'asciidoctor-pdf'
    "let g:asciidoctor_pdf_extensions = ['asciidoctor-diagram']
    let g:asciidoctor_pandoc_executable = 'pandoc'
    let g:asciidoctor_pandoc_data_dir = '~/Documents/.pandoc'
    "let g:asciidoctor_pandoc_other_params = '--toc'
    let g:asciidoctor_syntax_conceal = 1
    let g:asciidoctor_syntax_indented = 1
    let g:assciidoctor_fenced_languages = ['python', 'c' ]
    function! AsciidoctorMappings() 
        nnoremap <buffer> <leader>oo :AsciidoctorOpenRaw<CR>
        nnoremap <buffer> <leader>op :AsciidoctorOpenPDF<CR>
        nnoremap <buffer> <leader>oh :AsciidoctorOpenHTML<CR>
        nnoremap <buffer> <leader>ox :AsciidoctorOpenDOCX<CR>
        nnoremap <buffer> <leader>ch :Asciidoctor2HTML<CR>
        nnoremap <buffer> <leader>cp :Asciidoctor2PDF<CR>
        nnoremap <buffer> <leader>cx :Asciidoctor2DOCX<CR>
        nnoremap <buffer> <leader>p :AsciidoctorPasteImage<CR>
        ":make will build PDF's
        compiler asciidoctor2pdf
    endfunction
    "Call AsciidoctorMappings for all adoc type files. 
    augroup asciidoctor
        autocmd! 
        autocmd BufEnter *.adoc, *.asciidoc call AsciidoctorMappings()
    augroup END
```

**Comment by habamax** [comment]
Why do you have space again?

I am not 100% sure, but move asciidoctor settings out of plug#begin/plug#end

**Comment by KingJulianS** [comment]
Sorry I just recopied from the previous issue I logged in github. Hence the space.  
I have moved it outside, but still the same. 

```
    "below is for Haba asciidoctor
    let g:asciidoctor_executable = 'asciidoctor'
    "let g:asciidoctor_extensions = ['asciidoctor-diagram', 'asciidoctor-rouge']
    let g:asciidoctor_pdf_executable = 'asciidoctor-pdf'
    "let g:asciidoctor_pdf_extensions = ['asciidoctor-diagram']
    let g:asciidoctor_pandoc_executable = 'pandoc'
    let g:asciidoctor_pandoc_data_dir = '~/Documents/.pandoc'
    "let g:asciidoctor_pandoc_other_params = '--toc'
    let g:asciidoctor_syntax_conceal = 1
    let g:asciidoctor_syntax_indented = 1
    let g:assciidoctor_fenced_languages = ['python', 'c' ]

    function! AsciidoctorMappings() 
        nnoremap <buffer> <leader>oo :AsciidoctorOpenRaw<CR>
        nnoremap <buffer> <leader>op :AsciidoctorOpenPDF<CR>
        nnoremap <buffer> <leader>oh :AsciidoctorOpenHTML<CR>
        nnoremap <buffer> <leader>ox :AsciidoctorOpenDOCX<CR>
        nnoremap <buffer> <leader>ch :Asciidoctor2HTML<CR>
        nnoremap <buffer> <leader>cp :Asciidoctor2PDF<CR>
        nnoremap <buffer> <leader>cx :Asciidoctor2DOCX<CR>
        nnoremap <buffer> <leader>p :AsciidoctorPasteImage<CR>
        ":make will build PDF's
        compiler asciidoctor-pdf
    endfunction
    "Call AsciidoctorMappings for all adoc type files. 
    augroup asciidoctor
        autocmd! 
        autocmd BufEnter *.adoc,*.asciidoc call AsciidoctorMappings()
    augroup END
```

**Comment by KingJulianS** [comment]
I went on and tested anyway, since I saw that at least it set filetype to asciidoctor now (after space was removed) in the auto command.

* I could generate HTML and open it with (<leader>ch and oh). It was just ugly, but that's because I don't know CSS and don't have a custom CSS stylesheet to reference.
** HTML being nice by default is not an issue for me, since my HTML requirement is only to view. So I can use Firefox with the .adoc file directly, and it displays it nice enough using the asciidoc extension for Firefox. 

* I could generate PDF and open it with the Debian default PDF viewer(<leader>cp and co), and it looked fine. 

* I kept getting the following error msg when generating HTML and PDF, and assume it's related to this issue, wrt. The compiler not supported   : 
** asciidoctor: FAILED: 'asciidoctor-diagram' could not be loaded
  Use --trace for backtrace
